/*
 * Copyright (c) 2016 Cisco and/or its affiliates, and
 *                    Cable Television Laboratories, Inc. ("CableLabs")
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/*
 * File has been generated by proto_generator.py
 */

syntax = "proto2";

import "cfg.proto";

message t_SlaveDescriptor {
    required string addr_master =1 [default = "127.0.0.1"];
    required uint32 port_master =2 [default = 6000];
    required string interface_master =3 [default = "lo"];

    required string addr_local = 10 [default = "127.0.0.1"];
    required uint32 port_local =11 [default = 6100];
    required string interface_local = 12 [default = "lo"];

    required uint32 addr_family = 20 [default = 2];
}

message t_MasterSimApiMessage {
    optional t_SlaveDescriptor slave_descriptor = 1;
    enum t_Operation{
        READ = 1;
        WRITE = 2;
    }
    required t_Operation operation = 2;
    required config cfg = 3;
}
